src/main/java/org/web3j/mavenplugin/JavaClassGeneratorMojo.java

```diff
@@ -228,9 +227,6 @@ private String parseSoliditySources(Collection<String> includedFiles) throws Moj
                 SolidityCompiler.Options.METADATA
         );
         if (result.isFailed()) {
-            if (result.errors.contains("Source file requires different compiler version")) {
-                throw new VersionMismatchException(SolidityCompiler.getInstance(getLog()).getUsedSolCVersion(), result.errors);
-            }
             throw new MojoExecutionException("Could not compile solidity files\n" + result.errors);
```

Comment:
Maybe replace `solidity` by `Solidity` for clarity.

Reply:
Fixed